src/front/wgsl/mod.rs

```diff
@@ -158,9 +188,25 @@ impl<'a> Error<'a> {
                             }
                         }
                         ExpectedToken::Identifier => "identifier".to_string(),
-                        ExpectedToken::Float => "floating point literal".to_string(),
-                        ExpectedToken::Uint => "non-negative integer literal".to_string(),
-                        ExpectedToken::Sint => "integer literal".to_string(),
+                        ExpectedToken::Number { ty, width } => {
+                            let literal_ty_str = ty.map(|ty| match ty {
```

Comment:
can we do `match ty { Some(..) => "...", None => ".." }` instead of the `map`/`unwrap_or`?